packages/ses/src/get-anonymous-intrinsics.js

```diff
@@ -135,6 +118,33 @@ export const getAnonymousIntrinsics = () => {
     '%InertCompartment%': InertCompartment,
   };
 
+  try {
+    // 25.3.1 The AsyncGeneratorFunction Constructor
+
+    // eslint-disable-next-line no-empty-function
+    const AsyncGeneratorFunctionInstance = new FERAL_FUNCTION(
+      'return (async function* AsyncGeneratorFunctionInstance() {})',
+    )();
+    const AsyncGeneratorFunction = getConstructorOf(
+      AsyncGeneratorFunctionInstance,
+    );
+
+    // 25.3.2.2 AsyncGeneratorFunction.prototype
+    const AsyncGenerator = AsyncGeneratorFunction.prototype;
+    // 25.5.1 Properties of the AsyncGenerator Prototype Object
+    const AsyncGeneratorPrototype = AsyncGenerator.prototype;
+    const AsyncIteratorPrototype = getPrototypeOf(AsyncGeneratorPrototype);
+
+    assign(intrinsics, {
+      '%AsyncGenerator%': AsyncGenerator,
+      '%InertAsyncGeneratorFunction%': AsyncGeneratorFunction,
+      '%AsyncGeneratorPrototype%': AsyncGeneratorPrototype,
+      '%AsyncIteratorPrototype%': AsyncIteratorPrototype,
+    });
+  } catch {
+    // Silently swallow Hermes SyntaxError `async generators are unsupported` at runtime
+  }
```

Comment:
I would like to capture this in a narrower `try` that can only conceivably throw if async generators are unsupported, followed by an `if` block for the remaining behavior.

Reply:
This should also check that the error is a `SyntaxError`, and if not, rethrow it stopping ses initialization.

Never assume you know what a thrown error is without looking, or you'll mask other problems.

Reply:
> I would like to capture this in a narrower `try` that can only conceivably throw if async generators are unsupported, followed by an `if` block for the remaining behavior.

done ^ earlier suggestion now applied

Reply:
> This should also check that the error is a `SyntaxError`, and if not, rethrow it stopping ses initialization.
> 
> Never assume you know what a thrown error is without looking, or you'll mask other problems.

makes sense, i'll note relevant conditions here prior to making the change next

swallow
- `Error.prototype.jsEngine === 'hermes'`
- `e.name === 'SyntaxError'`
- `e.message.includes('async generators are unsupported')`

otherwise
- (re) `throw e;`

Reply:
added the `SyntaxError` check and rethrow to the `commons.js` construction in https://github.com/endojs/endo/commit/4b1b5e672477a7305e97601fcad26d48c186d141
which now looks good to use in other places